libs/api/src/test/java/de/cyface/api/MeasurementRetrieverTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2022 Cyface GmbH
+ *
+ * This file is part of the Cyface Data Collector.
+ *
+ * The Cyface Data Collector is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Cyface Data Collector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Cyface Data Collector. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.api;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import de.cyface.model.MeasurementIdentifier;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import de.cyface.apitestutils.TestMongoDatabase;
+import de.cyface.apitestutils.fixture.GeoLocationTestFixture;
+import de.cyface.apitestutils.fixture.TestMeasurementDocument;
+import io.vertx.core.Vertx;
+import io.vertx.ext.mongo.MongoClient;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+
+/**
+ * Tests the integration of {@link MeasurementRetriever} with MongoDB.
+ *
+ * @author Armin Schnabel
+ * @version 6.6.0
+ * @since 1.0.0
+ */
+@ExtendWith(VertxExtension.class)
+public class MeasurementRetrieverTest {
```

Comment:
This is similar to `MeasurementIteratorIntegrationTest` but one level more abstract, as `MeasurementRetriver` uses `MeasurementIterator`.